internal/handlers/common/aggregations/projection_iterator.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregations
+
+import (
+	"errors"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/iterator"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// ProjectionIterator returns an iterator that projects documents returned by the underlying iterator.
+// It will be added to the given closer.
+//
+// Next method returns the next projected document.
+//
+// Close method closes the underlying iterator.
+func ProjectionIterator(iter types.DocumentsIterator, closer *iterator.MultiCloser, projection *types.Document) (types.DocumentsIterator, error) { //nolint:lll // for readability
+	projectionValidated, inclusion, err := ValidateProjection(projection)
+	if errors.Is(err, errProjectionEmpty) {
```

Comment:
Minor comment: we use `switch-case` in other place likes this -- smth like

```switch {
case err == nil:
// ...
case errors.Is(...)
//
default:
// lazyerror
}
```

I suggest we follow the same approach everywhere, so the code looks more consistent and it's easier to read it.

Reply:
I personally prefer current implementation because `err == nil` looks less natural. Looking the codebase, we have mixture but it's sensible to decide which way we can do more consistently. @w84thesun, @noisersup, @AlekSi? :hugs:

Reply:
My personal preference would be to process errors in `switch` for more than 1 case in a way:
```
{
result, err := function()
if err != nil {
    switch {
    case errors.Is():
        // ...
    case errors.Is():
       // ...
    default:
      // ...
    }
}

// normal flow
}

Reply:
It seems that the number of prefered options will be equal to the number of people in the team :D

Reply:
Okay this one might not be easy to make it consistent. I'm realising I'm seeing more variety :see_no_evil: